tests/gnrc_tcp_server/main.c

```diff
@@ -72,6 +78,25 @@ int main(void)
     return 0;
 }
 
+bool verify_ip_addr(gnrc_netif_t *netif)
```

Comment:
Yes we need that. First of all I think it is okay to tolerate code duplication in test code sometimes. And this Code will be replaced anyway.

Secondly: IPv6 assigns link layer address to interfaces by default without any setup. That was the reason why I used them in the beginning. By using a different address type, I have to assign  to the server a valid non-link local Address. 

And because of the Issues we had, I want to make sure that the assigned Address is fully functional before I do anything else.

Reply:
Fine with the duplication for now... 

About the function itself: I do understand that you want to verify the address. **But** verification itself should not be the fix for the actual issue. If this was the case, I'd expect that some setup delay of the IPv6 address was causing the issue and with your verification, you just delay the process until the address was set up properly (or return false). Even that would be fine with me. But then I'd like to see that information in form of a comment.

Reply:
Verification is not the fix for the issue here it is a nice extra. Personally I would prefer to not care about the IP-Layer and its setup, because it is a test for the TCP-Layer. 

The verfication has nothing to do with the fix. The Issue seemed to be caused by the link local address that was used before. I'll try to fix the basic setup because some of the problems we encountered with sending "::" are caused by the test not by gnrc_tcp.

Reply:
Yes this has nothing to do with this and IMHO this function is unnecessary. When I said make sure it is valid I meant when you add it by setting the appropriate flags, not actually checking the validity with a function.